server/galaxyls/tests/files/complex_inputs_01_test.xml

```diff
@@ -0,0 +1,185 @@
+<test expect_num_outputs="$1">
+    <!--TODO: auto-generated test case. Please fill in the required values-->
+    <conditional name="c1">
+        <param name="c1_action" value="a1"/>
+        <param name="c1_a1_p1" value="$2"/>
+        <section name="sec1">
+            <param name="number" value="${3:1}"/>
+        </section>
+    </conditional>
+    <output name="sample">
+        <assert_contents>
+            <has_text text="$4"/>
+            <has_line line="$5"/>
+            <has_line_matching expression="$6"/>
+            <has_n_columns n="$7"/>
+            <has_size value="$8" delta="$9"/>
+        </assert_contents>
+    </output>
+    <output_collection name="out_collection" type="list">
+        <element name="$10">
+            <assert_contents>
+                <has_text text="$11"/>
+                <has_line line="$12"/>
+                <has_line_matching expression="$13"/>
+                <has_n_columns n="$14"/>
+                <has_size value="$15" delta="$16"/>
+            </assert_contents>
+        </element>
+    </output_collection>
+    <output_collection name="out_paired" type="paired">
+        <element name="$17">
+            <assert_contents>
+                <has_text text="$18"/>
+                <has_line line="$19"/>
+                <has_line_matching expression="$20"/>
+                <has_n_columns n="$21"/>
+                <has_size value="$22" delta="$23"/>
+            </assert_contents>
+        </element>
+    </output_collection>
+</test>
+
+<test expect_num_outputs="$24">
+    <!--TODO: auto-generated test case. Please fill in the required values-->
+    <conditional name="c1">
+        <param name="c1_action" value="a2"/>
+        <param name="c1_a2_p1" value="$25"/>
+        <conditional name="c2">
+            <param name="c2_action" value="c2a1"/>
+            <param name="c2_a1_p1" value="$26"/>
+            <section name="sec_c2a1">
+                <section name="sub_sec_c2a1">
+                    <param name="sub_p01" value="${27:3.2}"/>
+                    <param name="sub_p02" value="${28:4}"/>
```

Comment:
Cool, that is that ... a default value?

Reply:
Yes, the value after the `:` is the value defined in the `value` attribute (I think there was another `defaul_value` attribute but it is deprecated in the docs), this will appear as the value that you can replace (or leave it as is) when navigating through the placeholders ${X} with the `tab` key.